torch/distributions/constraints.py

```diff
@@ -463,13 +463,11 @@ class _PositiveDefinite(Constraint):
     event_dim = 2
 
     def check(self, value):
-        matrix_shape = value.shape[-2:]
+        # Assumes that matrix or matrices in value are symmetric.
+        # There are also non-symmetric positive definite matrices.
         batch_shape = value.unsqueeze(0).shape[:-2]
-        # TODO: replace with batched linear algebra routine when one becomes available
-        # note that `symeig()` returns eigenvalues in ascending order
-        flattened_value = value.reshape((-1,) + matrix_shape)
-        return torch.stack([v.symeig(eigenvectors=False)[0][:1] > 0.0
-                            for v in flattened_value]).view(batch_shape)
+        positive = torch.linalg.eigvalsh(value, UPLO="U")[..., :1] > 0.0
+        return positive.view(batch_shape)
```

Comment:
@IvanYashchuk could you double-check this one?

Reply:
I think here we can use `torch.linalg.cholesky_ex`.
```python
_, info = torch.linalg.cholesky_ex(value)
return info == 0
```
If `info` is larger than 0 then the input (symmetric) matrix is not positive definite numerically.

Reply:
I added it but left a small comment regarding the fact that we assume that the input matrix is symmetric, as before it just used the upper triangular part of the matrix and now we use the lower triangular part (which, btw, is not documented, (although perhaps it's better not to document it))